cmd/util/cmd/find-inconsistent-result/cmd.go

```diff
@@ -0,0 +1,164 @@
+package find_inconsistent_result
+
+import (
+	"fmt"
+
+	"github.com/dgraph-io/badger/v2"
+	"github.com/spf13/cobra"
+
+	"github.com/onflow/flow-go/cmd/util/cmd/common"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/state/protocol"
+	"github.com/onflow/flow-go/storage"
+)
+
+var (
+	flagDatadir   string
+	flagEndHeight uint64
+)
+
+var Cmd = &cobra.Command{
+	Use:   "find-inconsistent-result",
+	Short: "find the first block that produces inconsistent results",
+	Run:   run,
+}
+
+func init() {
+	Cmd.Flags().StringVarP(&flagDatadir, "datadir", "d", "/var/flow/data/protocol", "directory to the badger dababase")
+	_ = Cmd.MarkPersistentFlagRequired("datadir")
+
+	Cmd.Flags().Uint64Var(&flagEndHeight, "end-height", 0, "the last block height checks for result consistency")
+	_ = Cmd.MarkPersistentFlagRequired("end-height")
+}
+
+func run(*cobra.Command, []string) {
+	height, err := findFirstMismatch(flagDatadir, flagEndHeight)
+	if err != nil {
+		fmt.Printf("fatal: %v\n", err)
+		return
+	}
+
+	fmt.Printf("first mismatch found at height %v\n", height)
+}
+
+func findFirstMismatch(datadir string, endHeight uint64) (uint64, error) {
+	headers, results, state, db, err := createStorages(datadir)
+	defer db.Close()
+	if err != nil {
+		return 0, fmt.Errorf("could not create storages: %v", err)
+	}
+
+	c := &checker{
+		headers: headers,
+		results: results,
+		state:   state,
+	}
+
+	startHeight, err := findRootBlockHeight(state)
+	if err != nil {
+		return 0, fmt.Errorf("could not find root block height: %v", err)
+	}
```

Comment:
Could set end height to `state.Final().Head().Height` and remove `--end-height`. Not sure if there's any reason to need to pass in anything different via flag.